### PR TITLE
Switch checked example & addition of inner labels

### LINKED ISSUE #120: Switch: show an example using controlled component
Right now all the examples for the switch don't provide the "checked" attribute which makes the component uncontrolled. It would be nice to actually have a controlled switch in example.

### PR DESCRIPTION
**Pull request by elpiel**
- Resolves #120 
- Adds inner labels
- fixes the example to look like https://blueprintjs.com/docs/#core/components/switch

Question: How do I fix the inner label properties case?
They should be in `cameCase` however I didn't find a way to rename them using `yew`. One way would be to suppress the `#[warn(non_snake_case)]`.

### CODE REVIEW

File: yewprint/src/switch.rs [hunk 1]
```diff
@@ -44,6 +48,34 @@ impl Component for Switch {
...
+                                        inner_label
+                                    }
+                                }
+                            </div>
+                        </div>
+                        <div class=classes!("bp3-control-indicator-child")>
+                            <div class=classes!("bp3-switch-inner-text")>{inner_label}</div>
+                        </div>
+                    </>
+                }
```
Comment (cecton) [inline yewprint/src/switch.rs:72]:
```suggestion
            if self.props.inner_label.is_some() || self.props.inner_label_checked.is_some() {
                let inner_label = self.props.inner_label.as_ref().unwrap_or_default();
                let inner_label_checked = self.props.inner_label_checked.as_ref();
                html! {
                    <>
                        <div class=classes!("bp3-control-indicator-child")>
                            <div class=classes!("bp3-switch-inner-text")>
                                {
                                    if let Some(label_checked) = inner_label_checked {
                                        label_checked.clone()
                                    } else {
                                        inner_label.clone()
                                    }
                                }
                            </div>
                        </div>
                        <div class=classes!("bp3-control-indicator-child")>
                            <div class=classes!("bp3-switch-inner-text")>{inner_label.clone()}</div>
                        </div>
                    </>
                }
```

File: yewprint/src/switch.rs [hunk 2]
```diff
@@ -44,6 +48,34 @@ impl Component for Switch {
     }
 
     fn view(&self) -> Html {
+        let maybe_display_label = move || -> Html {
+            if self.props.inner_label.is_some() || self.props.inner_label_checked.is_some() {
+                let inner_label = self.props.inner_label.as_ref().unwrap_or_default();
```
Comment (cecton) [inline yewprint/src/switch.rs:53]:
```suggestion
                let inner_label = self.props.inner_label.as_deref().unwrap_or_default();
```

File: yewprint/src/switch.rs [hunk 3]
```diff
@@ -44,6 +48,34 @@ impl Component for Switch {
...
+                                    if let Some(label_checked) = inner_label_checked {
+                                        label_checked.clone()
+                                    } else {
+                                        inner_label.clone()
+                                    }
+                                }
+                            </div>
+                        </div>
+                        <div class=classes!("bp3-control-indicator-child")>
+                            <div class=classes!("bp3-switch-inner-text")>{inner_label.clone()}</div>
```
Comment (cecton) [inline yewprint/src/switch.rs:69]:
```suggestion
                                        inner_label.to_string()
                                    }
                                }
                            </div>
                        </div>
                        <div class=classes!("bp3-control-indicator-child")>
                            <div class=classes!("bp3-switch-inner-text")>{inner_label.to_string()}</div>
```

File: yewprint/src/switch.rs [hunk 4]
```diff
@@ -44,6 +48,34 @@ impl Component for Switch {
...
+                        </div>
+                        <div class=classes!("bp3-control-indicator-child")>
+                            <div class=classes!("bp3-switch-inner-text")>{inner_label.to_string()}</div>
+                        </div>
+                    </>
+                }
+            } else {
+                Html::default()
+            }
+        };
```
Comment (cecton) [inline yewprint/src/switch.rs:76]:
```suggestion
        let display_label = {
            if self.props.inner_label.is_some() || self.props.inner_label_checked.is_some() {
                let inner_label = self.props.inner_label.as_deref().unwrap_or_default();
                let inner_label_checked = self.props.inner_label_checked.as_ref();
                html! {
                    <>
                        <div class=classes!("bp3-control-indicator-child")>
                            <div class=classes!("bp3-switch-inner-text")>
                                {
                                    if let Some(label_checked) = inner_label_checked {
                                        label_checked.clone()
                                    } else {
                                        inner_label.to_string()
                                    }
                                }
                            </div>
                        </div>
                        <div class=classes!("bp3-control-indicator-child")>
                            <div class=classes!("bp3-switch-inner-text")>
                                {inner_label.to_string()}
                            </div>
                        </div>
                    </>
                }
            } else {
                Html::default()
            }
        };
```

File: yewprint/src/switch.rs [hunk 5]
```diff
@@ -64,6 +96,7 @@ impl Component for Switch {
             <span
                 class=classes!("bp3-control-indicator")
             >
+                {maybe_display_label()}
```
Comment (cecton) [inline yewprint/src/switch.rs:99]:
```suggestion
                {display_label}
```

File: yewprint/src/switch.rs [hunk 6]
```diff
@@ -44,6 +48,36 @@ impl Component for Switch {
...
+                            </div>
+                        </div>
+                    </>
+                }
+            } else {
+                Html::default()
+            }
+        };
+
+
```
Comment (cecton) [inline yewprint/src/switch.rs:80]:
```suggestion

```

**Comment by elpiel** [comment]
Thank you for checking out the PR @cecton !

**Comment by cecton** [comment]
Sorry for the late reply! [I got the notifications turned off for some reason.](https://twitter.com/CecileTonglet/status/1469965435160432640?s=20) Unfortunately I think this will be hard to reproduce on my own, can you check one of these solutions?

This PR looks great! I made a mess while trying to fix the issue :sweat_smile: sorry for that!

> Question: How do I fix the inner label properties case?
> They should be in `cameCase` however I didn't find a way to rename them using `yew`. One way would be to suppress the `#[warn(non_snake_case)]`.

We tend to convert everything in snake_case following the Rust style guidelines. It's not Yewprint's intention to make a 1:1 match API with blueprintjs

**Comment by cecton** [comment]
I have no clue why this CI refuses to deploy the site xD

```
Run nwtgck/actions-netlify@v1.1
  with:
    publish-dir: ./build
    github-token: ***
    deploy-message: Switch checked example & addition of inner labels
  env:
    CARGO_TERM_COLOR: always
    CACHE_ON_FAILURE: false
    CARGO_INCREMENTAL: 0
    NETLIFY_AUTH_TOKEN: 
    NETLIFY_SITE_ID: 
Netlify credentials not provided, not deployable
```

The secrets are set properly. Just use them god damn it

**Comment by cecton** [comment]
Ok overall the PR looks good and can be merged but you're not addressing at all #120

So I will merge as is and feel free to open another PR if you want to address #120

**Pull request merged**